### ISSUE TITLE
Crash on repository selection

### ISSUE THREAD

**Issue by zanchey** [gitx/gitx#170]
I've built from 9a99f1d0 and blown away my old GitX data in ~/Library, but GitX crashes when I open a repository:

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Illegal instruction: 4
Termination Reason:    Namespace SIGNAL, Code 0x4
Terminating Process:   exc handler [0]

Application Specific Information:
Crashing on exception: -[NSApplication effectiveAppearance]: unrecognized selector sent to instance 0x604000112990

Application Specific Backtrace 1:
0   CoreFoundation                      0x00007fff4b9e423b __exceptionPreprocess + 171
1   libobjc.A.dylib                     0x00007fff72c76c76 objc_exception_throw + 48
2   CoreFoundation                      0x00007fff4ba7cd94 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
3   CoreFoundation                      0x00007fff4b95a700 ___forwarding___ + 1456
4   CoreFoundation                      0x00007fff4b95a0c8 _CF_forwarding_prep_0 + 120
5   GitX                                0x0000000109f1f54c -[NSApplication(PBDarkMode) isDarkMode] + 44
6   GitX                                0x0000000109f2056a -[PBGitGradientBarView drawRect:] + 42
7   AppKit                              0x00007fff497a6f29 _NSViewDrawRect + 83
8   AppKit                              0x00007fff497b41c4 -[NSView(NSInternal) _recursive:displayRectIgnoringOpacity:inGraphicsContext:shouldChangeFontReferenceColor:] + 1578
9   AppKit                              0x00007fff497b3b94 -[NSView(NSInternal) _recursive:displayRectIgnoringOpacity:inContext:shouldChangeFontReferenceColor:] + 56
10  AppKit                              0x00007fff49045074 __46-[NSView(NSLayerKitGlue) drawLayer:inContext:]_block_invoke + 289
11  AppKit                              0x00007fff49044c08 -[NSView(NSLayerKitGlue) _drawViewBackingLayer:inContext:drawingHandler:] + 1577
12  AppKit                              0x00007fff490445d9 -[NSView(NSLayerKitGlue) drawLayer:inContext:] + 77
13  QuartzCore                          0x00007fff56ce0566 CABackingStoreUpdate_ + 300
14  QuartzCore                          0x00007fff56ce0433 ___ZN2CA5Layer8display_Ev_block_invoke + 53
15  QuartzCore                          0x00007fff56cdfcaa -[CALayer _display] + 2238
16  AppKit                              0x00007fff4904359e _NSBackingLayerDisplay + 556
17  AppKit                              0x00007fff49038409 -[_NSViewBackingLayer display] + 242
18  QuartzCore                          0x00007fff56cd0dc1 _ZN2CA5Layer17display_if_neededEPNS_11TransactionE + 633
19  QuartzCore                          0x00007fff56cd087f _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 35
20  QuartzCore                          0x00007fff56ccf880 _ZN2CA7Context18commit_transactionEPNS_11TransactionE + 326
21  QuartzCore                          0x00007fff56ccf425 _ZN2CA11Transaction6commitEv + 487
22  AppKit                              0x00007fff497efa72 __65+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayRefresh]_block_invoke + 283
23  CoreFoundation                      0x00007fff4b97c2f7 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
24  CoreFoundation                      0x00007fff4b97c21f __CFRunLoopDoObservers + 527
25  CoreFoundation                      0x00007fff4b95e798 __CFRunLoopRun + 1240
26  CoreFoundation                      0x00007fff4b95e033 CFRunLoopRunSpecific + 483
27  HIToolbox                           0x00007fff4ac48d96 RunCurrentEventLoopInMode + 286
28  HIToolbox                           0x00007fff4ac48a0f ReceiveNextEventCommon + 366
29  HIToolbox                           0x00007fff4ac48884 _BlockUntilNextEventMatchingListInModeWithFilter + 64
30  AppKit                              0x00007fff48ef8a73 _DPSNextEvent + 2085
31  AppKit                              0x00007fff4968ee34 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 3044
32  AppKit                              0x00007fff48eed885 -[NSApplication run] + 764
33  AppKit                              0x00007fff48ebca72 NSApplicationMain + 804
34  GitX                                0x0000000109f0fe0b main + 43
35  libdyld.dylib                       0x00007fff73890015 start + 1

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   com.apple.AppKit              	0x00007fff49223454 -[NSApplication _crashOnException:] + 109
1   com.apple.AppKit              	0x00007fff497efcb1 __65+[CATransaction(NSCATransaction) NS_setFlushesWithDisplayRefresh]_block_invoke + 858
2   com.apple.CoreFoundation      	0x00007fff4b97c2f7 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
3   com.apple.CoreFoundation      	0x00007fff4b97c21f __CFRunLoopDoObservers + 527
4   com.apple.CoreFoundation      	0x00007fff4b95e798 __CFRunLoopRun + 1240
5   com.apple.CoreFoundation      	0x00007fff4b95e033 CFRunLoopRunSpecific + 483
6   com.apple.HIToolbox           	0x00007fff4ac48d96 RunCurrentEventLoopInMode + 286
7   com.apple.HIToolbox           	0x00007fff4ac48a0f ReceiveNextEventCommon + 366
8   com.apple.HIToolbox           	0x00007fff4ac48884 _BlockUntilNextEventMatchingListInModeWithFilter + 64
9   com.apple.AppKit              	0x00007fff48ef8a73 _DPSNextEvent + 2085
10  com.apple.AppKit              	0x00007fff4968ee34 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 3044
11  com.apple.AppKit              	0x00007fff48eed885 -[NSApplication run] + 764
12  com.apple.AppKit              	0x00007fff48ebca72 NSApplicationMain + 804
13  net.phere.GitX                	0x0000000109f0fe0b main + 43 (main.m:15)
14  libdyld.dylib                 	0x00007fff73890015 start + 1

Thread 1:
0   libsystem_kernel.dylib        	0x00007fff739e128a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff73ba8009 _pthread_wqthread + 1035
2   libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 2:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff739d720a mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff739d6724 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff4b95f665 __CFRunLoopServiceMachPort + 341
3   com.apple.CoreFoundation      	0x00007fff4b95e9b7 __CFRunLoopRun + 1783
4   com.apple.CoreFoundation      	0x00007fff4b95e033 CFRunLoopRunSpecific + 483
5   com.apple.AppKit              	0x00007fff49035fc4 _NSEventThread + 184
6   libsystem_pthread.dylib       	0x00007fff73ba8661 _pthread_body + 340
7   libsystem_pthread.dylib       	0x00007fff73ba850d _pthread_start + 377
8   libsystem_pthread.dylib       	0x00007fff73ba7bf9 thread_start + 13

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff739e128a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff73ba8009 _pthread_wqthread + 1035
2   libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 4:: Dispatch queue: com.apple.root.user-interactive-qos
0   libsystem_kernel.dylib        	0x00007fff739d720a mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff739d6724 mach_msg + 60
2   com.apple.QuartzCore          	0x00007fff56ce72a3 CA::Render::Fence::wait(unsigned int, unsigned long) + 169
3   com.apple.QuartzCore          	0x00007fff56ccfc6c CA::Context::commit_transaction(CA::Transaction*) + 1330
4   com.apple.QuartzCore          	0x00007fff56ccf425 CA::Transaction::commit() + 487
5   com.apple.AppKit              	0x00007fff4901dc20 NSPerformVisuallyAtomicChange + 166
6   com.apple.AppKit              	0x00007fff4902a87b __42-[NSAnimation(NSInternal) _runInNewThread]_block_invoke + 85
7   libdispatch.dylib             	0x00007fff7385e5fa _dispatch_call_block_and_release + 12
8   libdispatch.dylib             	0x00007fff73856db8 _dispatch_client_callout + 8
9   libdispatch.dylib             	0x00007fff73858b2c _dispatch_root_queue_drain + 902
10  libdispatch.dylib             	0x00007fff73858755 _dispatch_worker_thread3 + 101
11  libsystem_pthread.dylib       	0x00007fff73ba8169 _pthread_wqthread + 1387
12  libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 5:: Dispatch queue: NSURLConnection-2
0   libsystem_kernel.dylib        	0x00007fff739e0a16 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff73ba9589 _pthread_cond_wait + 732
2   com.apple.Foundation          	0x00007fff4db24be0 -[__NSOperationInternal _waitUntilFinished:] + 896
3   com.apple.CFNetwork           	0x00007fff4a88df81 -[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:] + 197
4   com.apple.CFNetwork           	0x00007fff4a88deaa -[NSURLConnectionInternal _withActiveConnectionAndDelegate:] + 48
5   com.apple.CFNetwork           	0x00007fff4a891858 _NSURLConnectionDidReceiveResponse(_CFURLConnection*, _CFURLResponse*, void const*) + 81
6   com.apple.CFNetwork           	0x00007fff4a891772 invocation function for block in URLConnectionClient_Classic::_delegate_didReceiveResponse(_CFURLResponse*) + 110
7   com.apple.CFNetwork           	0x00007fff4aa033cf invocation function for block in URLConnectionClient_Classic::_withDelegateAsync(char const*, void (_CFURLConnection*, CFURLConnectionClientCurrent_VMax const*) block_pointer) + 100
8   com.apple.CFNetwork           	0x00007fff4a8c1ef3 invocation function for block in QCoreSchedulingSet::performAsync(void () block_pointer) const + 41
9   libdispatch.dylib             	0x00007fff7385e5fa _dispatch_call_block_and_release + 12
10  libdispatch.dylib             	0x00007fff73856db8 _dispatch_client_callout + 8
11  libdispatch.dylib             	0x00007fff7386b217 _dispatch_queue_serial_drain + 635
12  libdispatch.dylib             	0x00007fff7385e166 _dispatch_queue_invoke + 373
13  libdispatch.dylib             	0x00007fff7386bf0d _dispatch_root_queue_drain_deferred_wlh + 332
14  libdispatch.dylib             	0x00007fff7386fd21 _dispatch_workloop_worker_thread + 880
15  libsystem_pthread.dylib       	0x00007fff73ba7fd2 _pthread_wqthread + 980
16  libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 6:: Dispatch queue: NSOperationQueue 0x6000006335c0 (QOS: UNSPECIFIED)
0   libsystem_kernel.dylib        	0x00007fff739d7246 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff73866893 _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff7385e979 _dispatch_semaphore_wait_slow + 101
3   com.apple.WebCore             	0x00007fff58ce1c74 -[WebCoreResourceHandleAsOperationQueueDelegate connection:didReceiveResponse:] + 180
4   com.apple.CFNetwork           	0x00007fff4a88e05c __65-[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:]_block_invoke + 53
5   com.apple.Foundation          	0x00007fff4da7752f __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
6   com.apple.Foundation          	0x00007fff4da77391 -[NSBlockOperation main] + 68
7   com.apple.Foundation          	0x00007fff4da7583e -[__NSOperationInternal _start:] + 778
8   com.apple.Foundation          	0x00007fff4da71867 __NSOQSchedule_f + 369
9   libdispatch.dylib             	0x00007fff73856db8 _dispatch_client_callout + 8
10  libdispatch.dylib             	0x00007fff73869e81 _dispatch_continuation_pop + 472
11  libdispatch.dylib             	0x00007fff73861733 _dispatch_async_redirect_invoke + 703
12  libdispatch.dylib             	0x00007fff738589a9 _dispatch_root_queue_drain + 515
13  libdispatch.dylib             	0x00007fff73858755 _dispatch_worker_thread3 + 101
14  libsystem_pthread.dylib       	0x00007fff73ba8169 _pthread_wqthread + 1387
15  libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 7:
0   libsystem_kernel.dylib        	0x00007fff739e0a16 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff73ba9589 _pthread_cond_wait + 732
2   libc++.1.dylib                	0x00007fff717e4d43 std::__1::condition_variable::__do_timed_wait(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::system_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) + 93
3   com.apple.JavaScriptCore      	0x00007fff4f5d5ab5 std::__1::cv_status std::__1::condition_variable::wait_until<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >(std::__1::unique_lock<std::__1::mutex>&, std::__1::chrono::time_point<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > > const&) + 117
4   com.apple.JavaScriptCore      	0x00007fff4f5d59a0 std::__1::cv_status std::__1::condition_variable_any::wait_until<std::__1::unique_lock<bmalloc::Mutex>, std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >(std::__1::unique_lock<bmalloc::Mutex>&, std::__1::chrono::time_point<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > > const&) + 96
5   com.apple.JavaScriptCore      	0x00007fff4f5d55c9 bmalloc::Scavenger::threadRunLoop() + 361
6   com.apple.JavaScriptCore      	0x00007fff4f5d5359 bmalloc::Scavenger::threadEntryPoint(bmalloc::Scavenger*) + 9
7   com.apple.JavaScriptCore      	0x00007fff4f5d5698 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(bmalloc::Scavenger*), bmalloc::Scavenger*> >(void*) + 40
8   libsystem_pthread.dylib       	0x00007fff73ba8661 _pthread_body + 340
9   libsystem_pthread.dylib       	0x00007fff73ba850d _pthread_start + 377
10  libsystem_pthread.dylib       	0x00007fff73ba7bf9 thread_start + 13

Thread 8:: Dispatch queue: NSURLConnection-1
0   libsystem_kernel.dylib        	0x00007fff739e0a16 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff73ba9589 _pthread_cond_wait + 732
2   com.apple.Foundation          	0x00007fff4db24be0 -[__NSOperationInternal _waitUntilFinished:] + 896
3   com.apple.CFNetwork           	0x00007fff4a88df81 -[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:] + 197
4   com.apple.CFNetwork           	0x00007fff4a88deaa -[NSURLConnectionInternal _withActiveConnectionAndDelegate:] + 48
5   com.apple.CFNetwork           	0x00007fff4a891858 _NSURLConnectionDidReceiveResponse(_CFURLConnection*, _CFURLResponse*, void const*) + 81
6   com.apple.CFNetwork           	0x00007fff4a891772 invocation function for block in URLConnectionClient_Classic::_delegate_didReceiveResponse(_CFURLResponse*) + 110
7   com.apple.CFNetwork           	0x00007fff4aa033cf invocation function for block in URLConnectionClient_Classic::_withDelegateAsync(char const*, void (_CFURLConnection*, CFURLConnectionClientCurrent_VMax const*) block_pointer) + 100
8   com.apple.CFNetwork           	0x00007fff4a8c1ef3 invocation function for block in QCoreSchedulingSet::performAsync(void () block_pointer) const + 41
9   libdispatch.dylib             	0x00007fff7385e5fa _dispatch_call_block_and_release + 12
10  libdispatch.dylib             	0x00007fff73856db8 _dispatch_client_callout + 8
11  libdispatch.dylib             	0x00007fff7386b217 _dispatch_queue_serial_drain + 635
12  libdispatch.dylib             	0x00007fff7385e166 _dispatch_queue_invoke + 373
13  libdispatch.dylib             	0x00007fff7386bf0d _dispatch_root_queue_drain_deferred_wlh + 332
14  libdispatch.dylib             	0x00007fff7386fd21 _dispatch_workloop_worker_thread + 880
15  libsystem_pthread.dylib       	0x00007fff73ba7fd2 _pthread_wqthread + 980
16  libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 9:: Dispatch queue: NSOperationQueue 0x6000006335c0 (QOS: UNSPECIFIED)
0   libsystem_kernel.dylib        	0x00007fff739d7246 semaphore_wait_trap + 10
1   libdispatch.dylib             	0x00007fff73866893 _dispatch_sema4_wait + 16
2   libdispatch.dylib             	0x00007fff7385e979 _dispatch_semaphore_wait_slow + 101
3   com.apple.WebCore             	0x00007fff58ce1c74 -[WebCoreResourceHandleAsOperationQueueDelegate connection:didReceiveResponse:] + 180
4   com.apple.CFNetwork           	0x00007fff4a88e05c __65-[NSURLConnectionInternal _withConnectionAndDelegate:onlyActive:]_block_invoke + 53
5   com.apple.Foundation          	0x00007fff4da7752f __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
6   com.apple.Foundation          	0x00007fff4da77391 -[NSBlockOperation main] + 68
7   com.apple.Foundation          	0x00007fff4da7583e -[__NSOperationInternal _start:] + 778
8   com.apple.Foundation          	0x00007fff4da71867 __NSOQSchedule_f + 369
9   libdispatch.dylib             	0x00007fff73856db8 _dispatch_client_callout + 8
10  libdispatch.dylib             	0x00007fff73869e81 _dispatch_continuation_pop + 472
11  libdispatch.dylib             	0x00007fff73861733 _dispatch_async_redirect_invoke + 703
12  libdispatch.dylib             	0x00007fff738589a9 _dispatch_root_queue_drain + 515
13  libdispatch.dylib             	0x00007fff73858755 _dispatch_worker_thread3 + 101
14  libsystem_pthread.dylib       	0x00007fff73ba8169 _pthread_wqthread + 1387
15  libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 10:
0   libsystem_kernel.dylib        	0x00007fff739e128a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff73ba8009 _pthread_wqthread + 1035
2   libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 11:
0   libsystem_kernel.dylib        	0x00007fff739e128a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff73ba820e _pthread_wqthread + 1552
2   libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 12:
0   libsystem_kernel.dylib        	0x00007fff739e128a __workq_kernreturn + 10
1   libsystem_pthread.dylib       	0x00007fff73ba8009 _pthread_wqthread + 1035
2   libsystem_pthread.dylib       	0x00007fff73ba7be9 start_wqthread + 13

Thread 13:
0   libsystem_pthread.dylib       	0x00007fff73ba7bdc start_wqthread + 0

Thread 14:
0   libsystem_pthread.dylib       	0x00007fff73ba7bdc start_wqthread + 0
1   ???                           	0x0000000000050000 0 + 327680

Thread 15:
0   libsystem_pthread.dylib       	0x00007fff73ba7bdc start_wqthread + 0
1   ???                           	0x0001000000030000 0 + 281474976907264

Thread 16:
0   libsystem_pthread.dylib       	0x00007fff73ba7bdc start_wqthread + 0

Thread 17:
0   libsystem_pthread.dylib       	0x00007fff73ba7bdc start_wqthread + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x00007f8cea018a00  rbx: 0x00006040008444d0  rcx: 0x0000000000000000  rdx: 0x0000000000000e8f
  rdi: 0x0000000008000101  rsi: 0x00007f8cea0d0261  rbp: 0x00007ffee5d5fe30  rsp: 0x00007ffee5d5fe20
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x00007ffee5d5f928  r11: 0xfffffffffff495ff
  r12: 0x000060400012ac80  r13: 0x00007f8ce9803050  r14: 0x00007fff72c68e80  r15: 0x00007fff4b97c5b0
  rip: 0x00007fff49223454  rfl: 0x0000000000010202  cr2: 0x0000000109fa27eb
```

Built on macOS 10.13.6 with Xcode 10.1.

**Comment by tiennou** [comment]
That should be fixed by 52561d3bf1b78d41fea566dd563f8def37a69c2c.

**Comment by zanchey** [comment]
Yes, it does!
